soc/usbexecd.c

```diff
@@ -0,0 +1,1029 @@
+#define _GNU_SOURCE
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <errno.h>
+#include <poll.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <sys/wait.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <syslog.h>
+#include <stdlib.h>
+#include <sys/epoll.h>
+#include <sys/signalfd.h>
+
+enum Commands {
+    CMD_RESET = 0x0,
+    CMD_OPEN = 0x1,
+    CMD_CLOSE = 0x2,
+    CMD_KILL = 0x3,
+    CMD_EXIT_STATUS = 0x5,
+    CMD_CLOSE_ACK = 0x6,
+
+    CMD_WRITE_CONTROL = 0x10,
+    CMD_WRITE_STDIN = 0x11,
+    CMD_WRITE_STDOUT = 0x12,
+    CMD_WRITE_STDERR = 0x13,
+
+    CMD_ACK_CONTROL = 0x20,
+    CMD_ACK_STDIN = 0x21,
+    CMD_ACK_STDOUT = 0x22,
+    CMD_ACK_STDERR = 0x23,
+
+    CMD_CLOSE_CONTROL = 0x30,
+    CMD_CLOSE_STDIN = 0x31,
+    CMD_CLOSE_STDOUT = 0x32,
+    CMD_CLOSE_STDERR = 0x33,
+};
+
+#define debug(args...) syslog(LOG_INFO, args)
```

Comment:
If you want to leave in `debug` calls that might be useful later, you can `#define debug(args...) /*nothing*/` here and they'll be compiled out.